### PR TITLE
MM-18379 Migrate tests from "app/plugin_api_test.go" to use testify

### PR DESCRIPTION
**Pull request by Lumexralph**
#### Summary

Modify the calls to t.Fatal and the condition checks that lead to them, replacing them with calls to the testify require or assert
#### Ticket Link

[MM-18379](https://mattermost.atlassian.net/browse/MM-18379)
Fixes https://github.com/mattermost/mattermost-server/issues/12102

### CODE REVIEW

File: app/plugin_api_test.go [hunk 1]
```diff
@@ -340,13 +340,11 @@ func TestPluginAPISavePluginConfig(t *testing.T) {
...
-	if err := json.Unmarshal([]byte(pluginConfigJsonString), &pluginConfig); err != nil {
-		t.Fatal(err)
-	}
+	err := json.Unmarshal([]byte(pluginConfigJsonString), &pluginConfig)
+	require.Nil(t, err)
 
-	if err := api.SavePluginConfig(pluginConfig); err != nil {
-		t.Fatal(err)
-	}
+	err = api.SavePluginConfig(pluginConfig)
```
Comment (hanzei) [inline app/plugin_api_test.go]: This is type `*model.AppError`. Please use a `appErr` as a variable name.

File: app/plugin_api_test.go [hunk 2]
```diff
@@ -340,13 +340,11 @@ func TestPluginAPISavePluginConfig(t *testing.T) {
 	pluginConfigJsonString := `{"mystringsetting": "str", "MyIntSetting": 32, "myboolsetting": true}`
 
 	var pluginConfig map[string]interface{}
-	if err := json.Unmarshal([]byte(pluginConfigJsonString), &pluginConfig); err != nil {
-		t.Fatal(err)
-	}
+	err := json.Unmarshal([]byte(pluginConfigJsonString), &pluginConfig)
+	require.Nil(t, err)
```
Comment (hanzei) [inline app/plugin_api_test.go]: Please use 
```suggestion
	require.NoError(t, err)
```

To be more explicit.

File: app/plugin_api_test.go [hunk 3]
```diff
@@ -355,14 +353,12 @@ func TestPluginAPISavePluginConfig(t *testing.T) {
...
-	}
+	err = api.LoadPluginConfiguration(savedConfiguration)
+	require.Nil(t, err)
 
 	expectedConfiguration := new(Configuration)
-	if err := json.Unmarshal([]byte(pluginConfigJsonString), &expectedConfiguration); err != nil {
-		t.Fatal(err)
-	}
+	err = json.Unmarshal([]byte(pluginConfigJsonString), &expectedConfiguration)
+	require.Nil(t, err)
```
Comment (hanzei) [inline app/plugin_api_test.go]:
```suggestion
	require.NoError(t, err)
```

File: app/plugin_api_test.go [hunk 4]
```diff
@@ -387,9 +383,9 @@ func TestPluginAPIGetPluginConfig(t *testing.T) {
 	pluginConfigJsonString := `{"mystringsetting": "str", "myintsetting": 32, "myboolsetting": true}`
 	var pluginConfig map[string]interface{}
 
-	if err := json.Unmarshal([]byte(pluginConfigJsonString), &pluginConfig); err != nil {
-		t.Fatal(err)
-	}
+	err := json.Unmarshal([]byte(pluginConfigJsonString), &pluginConfig)
+	require.Nil(t, err)
```
Comment (hanzei) [inline app/plugin_api_test.go]:
```suggestion
	require.NoError(t, err)
```

File: app/plugin_api_test.go [hunk 5]
```diff
@@ -403,9 +399,9 @@ func TestPluginAPILoadPluginConfiguration(t *testing.T) {
 	defer th.TearDown()
 
 	var pluginJson map[string]interface{}
-	if err := json.Unmarshal([]byte(`{"mystringsetting": "str", "MyIntSetting": 32, "myboolsetting": true}`), &pluginJson); err != nil {
-		t.Fatal(err)
-	}
+	err := json.Unmarshal([]byte(`{"mystringsetting": "str", "MyIntSetting": 32, "myboolsetting": true}`), &pluginJson)
+	require.Nil(t, err)
```
Comment (hanzei) [inline app/plugin_api_test.go]:
```suggestion
	require.NoError(t, err)
```

File: app/plugin_api_test.go [hunk 6]
```diff
@@ -474,9 +470,9 @@ func TestPluginAPILoadPluginConfigurationDefaults(t *testing.T) {
 	defer th.TearDown()
 
 	var pluginJson map[string]interface{}
-	if err := json.Unmarshal([]byte(`{"mystringsetting": "override"}`), &pluginJson); err != nil {
-		t.Fatal(err)
-	}
+	err := json.Unmarshal([]byte(`{"mystringsetting": "override"}`), &pluginJson)
+	require.Nil(t, err)
```
Comment (hanzei) [inline app/plugin_api_test.go]:
```suggestion
	require.NoError(t, err)
```

File: app/plugin_api_test.go (hunk 1 above)
Comment (Lumexralph) [reply app/plugin_api_test.go]: @hanzei Thanks.

File: app/plugin_api_test.go (hunk 6 above)
Comment (Lumexralph) [reply app/plugin_api_test.go]: @hanzei The suggested feedback has been implemented, thanks.

**Comment by hanzei** [comment]
@Lumexralph Please do not force push. It makes reviews harder, as the reviewer can't just look at the diff. See https://developers.mattermost.com/blog/submitting-great-prs/#4-avoid-force-pushing for more context.

**Comment by hanzei** [comment]
And please merge `master` into your branch. Let me know, if you need help.

**Comment by Lumexralph** [comment]
> @Lumexralph Please do not force push. It makes reviews harder, as the reviewer can't just look at the diff. See https://developers.mattermost.com/blog/submitting-great-prs/#4-avoid-force-pushing for more context.

@hanzei Thank you, I'm sorry for any inconvenience and that was very enlightening.

**Comment by Lumexralph** [comment]
> And please merge `master` into your branch. Let me know, if you need help.

@hanzei I want to be sure about this, what I have in mind is to update my local master branch with the remote master branch by `git rebase` and merge it into my current branch or is there another way I can approach this?

**Comment by hanzei** [comment]
> > And please merge `master` into your branch. Let me know, if you need help.
> 
> @hanzei I want to be sure about this, what I have in mind is to update my local master branch with the remote master branch by `git rebase` and merge it into my current branch or is there another way I can approach this?

This way is fine. If you want to follow a less "aggressive" way, you can merge you upstream branch with your local branch. Then no force pushing is required. See https://help.github.com/en/articles/syncing-a-fork

**Comment by Lumexralph** [comment]
@hanzei Please find time to review if your requested changes have been implemented.

**Pull request merged**